gdx/src/com/badlogic/gdx/math/FloatCounter.java

```diff
@@ -50,6 +50,15 @@ public FloatCounter (int windowSize) {
 	/** Add a value and update all fields.
 	 * @param value The value to add */
 	public void put (float value) {
+		if (Float.isNaN(value)) throw new RuntimeException("value is not a number");
+		/*
+		 * Suggestions that I make to avoid errors when putting big numbers :
+		 * 
+		 * else if (Float.isInfinite(value)) throw new RuntimeException("value is infinite"); else if
+		 * (value==Float.MAX_VALUE||value==-Float.MAX_VALUE) throw new RuntimeException("value is too big and " +
+		 * "there are risks that the next results will be different than expected");
```

Comment:
I do not like the idea of throwing an exception for crazy values. Someone might want to do that, or there might be some scenario where a game ends up at `MAX_VALUE` and now it'll crash (which is worse than just behaving incorrectly).

Reply:
should i keep the NaN exception though ?

Reply:
or maybe I should avoid having both POSITIVE_INFINITY and - NEGATIVE_INFINITY ? that's the issue I tested where the total was then NaN

Reply:
I wouldn't throw an exception over `put`ting a NaN value for much the same reason. Maybe we should return early on NaN to avoid screwing up the total/average. I'd be interested to hear what someone else thinks.

Infinities and other extreme values will always cause problems.

Reply:
I haven't inspected it but I reckon NaN would break `WindowedMean`.

Reply:
hey so should I keep FloatCounter as it is and rewrite the tests for it to take into account NaN etc ?

Reply:
If you want to. I'm still not that sure on any of this; would like to see if someone who actually uses `FloatCounter` has thoughts on the matter.

Reply:
okay, I can still wait for someone to confirm how it works

Reply:
Imagine if we checked for odd float values for _every_ method that accepted floats -- it isn't reasonable to guard against unless there are some special requirements.